Lib/multiprocessing/shared_memory.py

```diff
@@ -1,225 +1,231 @@
-"Provides shared memory for direct access across processes."
+"""Provides shared memory for direct access across processes.
 
+The API of this package is currently provisional. Refer to the
+documentation for details.
+"""
 
-__all__ = [ 'SharedMemory', 'PosixSharedMemory', 'WindowsNamedSharedMemory',
-            'ShareableList', 'shareable_wrap',
-            'SharedMemoryServer', 'SharedMemoryManager', 'SharedMemoryTracker' ]
 
+__all__ = [ 'SharedMemory', 'ShareableList' ]
 
-from functools import reduce
+
+from functools import partial
 import mmap
-from .managers import DictProxy, SyncManager, Server
-from . import util
 import os
-import random
+import errno
 import struct
-import sys
-try:
-    from _posixshmem import _PosixSharedMemory, Error, ExistentialError, O_CREX
-except ImportError as ie:
-    if os.name != "nt":
-        # On Windows, posixshmem is not required to be available.
-        raise ie
-    else:
-        _PosixSharedMemory = object
-        class ExistentialError(BaseException): pass
-        class Error(BaseException): pass
-        O_CREX = -1
-
-
-class WindowsNamedSharedMemory:
-
-    def __init__(self, name, flags=None, mode=None, size=None, read_only=False):
-        if name is None:
-            name = f'wnsm_{os.getpid()}_{random.randrange(100000)}'
-
-        self._mmap = mmap.mmap(-1, size, tagname=name)
-        self.buf = memoryview(self._mmap)
-        self.name = name
-        self.size = size
-
-    def __repr__(self):
-        return f'{self.__class__.__name__}({self.name!r}, size={self.size})'
-
-    def close(self):
-        self.buf.release()
-        self._mmap.close()
+import secrets
 
-    def unlink(self):
-        """Windows ensures that destruction of the last reference to this
-        named shared memory block will result in the release of this memory."""
-        pass
+if os.name == "nt":
+    import _winapi
+    _USE_POSIX = False
+else:
+    import _posixshmem
+    _USE_POSIX = True
 
 
-class PosixSharedMemory(_PosixSharedMemory):
+O_CREX = os.O_CREAT | os.O_EXCL
```

Comment:
Consider making this private (`_O_CREX`)

Reply:
Good point.  Changed to private in commit 12c097d88e7d1ceb159a1b726f229654e6218eed.